Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject
+        /// </summary>
+        /// <param name="go"></param>
+        protected virtual void OnTap(GameObject go)
```

Comment:
I thought about that, but I wanted to include the examples of usage in the methods to help new contributors understand how to use and implement in their own classes.